hub/util/version_control.py

```diff
@@ -218,13 +221,46 @@ def discard_old_metas(
             pass
 
 
+def _update_version_info(old, new):
+    """Merges 2 version infos by updating `old` version info inplace"""
+    old_commit_node_map = old["commit_node_map"]
+    new_commit_node_map = new["commit_node_map"]
+    for commit, node in new_commit_node_map.items():
```

Comment:
short comments describing the 3 loops would be helpful for people new to the codebase